components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/listener/TenantSyncListener.java

```diff
@@ -0,0 +1,401 @@
+    public static class EventRunner implements Runnable {
+
+        private String notificationEndpoint;
+        private String username;
+        private char[] password;
+        private Map<String, String> headers;
+        private TenantManagementEventDTO event;
+        private static final int MAX_RETRIES = 5;
+        private static final long RETRY_DELAY_MS = 5000L;
+
+        public EventRunner(String notificationEndpoint, String username, char[] password, Map<String, String> headers,
+                           TenantManagementEventDTO event) {
+
+            this.notificationEndpoint = notificationEndpoint;
+            this.username = username;
+            this.password = password;
+            this.headers = headers;
+            this.event = event;
+        }
+
+        @Override
+        public void run() {
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Sending HTTP request to notification endpoint: " + notificationEndpoint);
+            }
+
+            HttpPost httpPost = new HttpPost(notificationEndpoint);
+            if (StringUtils.isNotEmpty(username) && !(password == null || password.length == 0)) {
+                byte[] credentials = Base64
+                        .encodeBase64((username + ":" + new String(password)).getBytes(StandardCharsets.UTF_8));
+                httpPost.addHeader("Authorization", "Basic " + new String(credentials, StandardCharsets.UTF_8));
+            }
+
+            headers.forEach((key, value) -> {
+                httpPost.addHeader(key, value);
+            });
+
+            String content = new Gson().toJson(event);
+            httpPost.setEntity(new StringEntity(content, ContentType.APPLICATION_JSON));
+
+            for (int attempt = 1; attempt <= MAX_RETRIES; attempt++) {
+
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Sending notification... (Attempt " + attempt + "/" + MAX_RETRIES + ")");
+                }
+
+                try (CloseableHttpClient httpClient = HTTPClientUtils.createClientWithCustomHostnameVerifier()
+                        .build()) {
+
+                    boolean success = httpClient.execute(httpPost, response -> {
+                        if (response.getCode() != HttpStatus.SC_OK) {
+                            LOG.error("Error while notifying for tenant creation. Status code: " + response.getCode());
+                            return false;
+                        }
+                        return true;
+                    });
+
+                    if (success) {
+                        return;
+                    }
+
+                } catch (IOException e) {
+                    LOG.error("An error occurred while sending the HTTP request: ", e);
+                }
+
+                if (attempt < MAX_RETRIES) {
+                    try {
+                        LOG.info("Will retry sending notification in " + RETRY_DELAY_MS + "ms.");
+                        Thread.sleep(RETRY_DELAY_MS);
```

Comment:
A better approach is to increase the delay for each retry incrementally.
Starting with 500ms.

If the issue is temporary and caused by high load, hitting the service again and again at the same fixed rate can make the problem worse.

Reply:
Improved the retry logic in https://github.com/wso2/carbon-identity-framework/pull/7075/commits/272f7a1cb9027d08ff3e8223a88c5a4eaa4da693